### DISCUSSION TITLE
Unable to connect to GCP Postgres with SSL enabled

### DISCUSSION THREAD

**Discussion by antongildebrand** [prisma/prisma#13419] [Q&A]
Hi!

I've been struggling with connecting Prisma to our Postgres database. We have SSL enabled, and a public IP that I try to connect to. When I disable the SSL requirement, I'm able to connect but with SSL enabled I get the error
```
Error opening a TLS connection: One or more parameters passed to a function were not valid.
``` 

I'm using a connection string like
```
postgresql://username:password@10.20.30.40/db?schema=public&sslidentity=postgres.p12&sslcert=server-ca.pem
```
The `server-ca.pem` file comes directly from GCP and I've tried rotating the server certificate as well. The `postgres.p12` file was created from the `client-cert.pem` and `client-key.pem` file using the command
```
openssl pkcs12 -export -out postgres.p12 -inkey client-key.pem -in client-cert.pem
```
No password was set during the creation of the PKCS12 file, but I've tried adding that as well without any luck (of course I added the password to the connection string as well then).

I've experimented with changing the `sslmode` and `sslaccept` parameters of the connection string as well without any luck.

## Versions
* PostgreSQL 14
* Prisma 3.14.0
* Node v16.15.0
* OpenSSL 3.0.3 used for generation of PKCS12 file.
* MacOS 11.6.2

How can I solve this error?

**Comment by janpio** [comment]
Here is an example of where we are testing this exact case: https://github.com/prisma/ecosystem-tests/tree/dev/databases/gcp-postgresql-ssl

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
